src/sagas/publicChannels/publicChannels.selectors.ts

```diff
@@ -130,12 +130,102 @@ export const currentChannelDisplayableMessages = createSelector(
         id: message.id,
         type: message.type,
         message: message.message,
-        createdAt: formatMessageDisplayDate(message.createdAt),
+        createdAt: message.createdAt,
         nickname: user.username,
       };
     })
 );
 
+// returns array of messages SORTED by createdAt
+export const currentChannelMessagesSortedByTime = createSelector(
+  currentChannelDisplayableMessages,
+  (messages) =>
+    messages.sort((a, b) => b.createdAt - a.createdAt)
+);
+
+// returns array of messages SORTED and GROUPED by createdAt and user
+export const currentChannelMessagesGroupedByUserTime = createSelector(
+  currentChannelMessagesSortedByTime,
+  (messages) => {
+    const timeOfGroupingMessages = 300
+    let newMessages = []
+
+    for (let indexOfMessages = 0; indexOfMessages < messages.length; indexOfMessages++) {
+      let currentMessage = messages[indexOfMessages]
+
+      if (messages[indexOfMessages + 1]) {
+        while ((currentMessage.createdAt - messages[indexOfMessages + 1].createdAt) < timeOfGroupingMessages
+          && currentMessage.nickname === messages[indexOfMessages + 1].nickname) {
+          currentMessage = {
+            ...currentMessage,
+            message: currentMessage.message + "\n" + messages[indexOfMessages + 1].message
```

Comment:
So we're still combining them into the same message? This seems like it will be a problem at some point. Is there a way we can add a flag to every message in the group (except the first) like 'grouped' that will change how it displays? If this is easy shouldn't we do it now?

Reply:
So the point of the selectors is to pass messages in displayable format to the frontend. The messages will remain untouched in db and also we'll be able to access them ungrouped with the help of other selectors